<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 6 Making measurements and comparisons | A Guide to YASARA</title>
  <meta name="description" content="A guide for the visualization program YASARA and the uses of the program">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 6 Making measurements and comparisons | A Guide to YASARA" />
  <meta property="og:type" content="book" />
  
  <meta property="og:image" content="images/cover.png" />
  <meta property="og:description" content="A guide for the visualization program YASARA and the uses of the program" />
  <meta name="github-repo" content="https://github.com/CEBerndsen/YASARA-guide" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Making measurements and comparisons | A Guide to YASARA" />
  
  <meta name="twitter:description" content="A guide for the visualization program YASARA and the uses of the program" />
  <meta name="twitter:image" content="images/cover.png" />

<meta name="author" content="Chris Berndsen">


<meta name="date" content="2019-01-14">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="adding-annotations-to-your-scene.html">
<link rel="next" href="movies.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A guide to biomolecule visualization and analysis in YASARA</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Index</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#why-read-this-book"><i class="fa fa-check"></i>Why read this book</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#what-is-in-this-guide"><i class="fa fa-check"></i>What is in this guide?</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#a-note-on-using-the-command-line"><i class="fa fa-check"></i>A note on using the command line</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#end-of-section-checks-and-active-reading"><i class="fa fa-check"></i>End of section checks and active reading</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#questions-and-corrections"><i class="fa fa-check"></i>Questions and corrections</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#copyright"><i class="fa fa-check"></i>Copyright</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#acknowledgements"><i class="fa fa-check"></i>Acknowledgements</a><ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#who-owns-yasara-and-the-other-programs-in-this-book"><i class="fa fa-check"></i>Who owns YASARA and the other programs in this book</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="1" data-path="obtaining-yasara.html"><a href="obtaining-yasara.html"><i class="fa fa-check"></i><b>1</b> Obtaining YASARA</a><ul>
<li class="chapter" data-level="1.1" data-path="obtaining-yasara.html"><a href="obtaining-yasara.html#what-is-yasara"><i class="fa fa-check"></i><b>1.1</b> What is YASARA</a></li>
<li class="chapter" data-level="1.2" data-path="obtaining-yasara.html"><a href="obtaining-yasara.html#download-yasaraview"><i class="fa fa-check"></i><b>1.2</b> Download YASARA<sub>view</sub></a></li>
<li class="chapter" data-level="1.3" data-path="obtaining-yasara.html"><a href="obtaining-yasara.html#installing-yasara"><i class="fa fa-check"></i><b>1.3</b> Installing YASARA</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html"><i class="fa fa-check"></i><b>2</b> The YASARA user interface</a><ul>
<li class="chapter" data-level="2.1" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#HUD"><i class="fa fa-check"></i><b>2.1</b> Heads Up Display (HUD)</a><ul>
<li class="chapter" data-level="2.1.1" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#atom-properties"><i class="fa fa-check"></i><b>2.1.1</b> Atom Properties</a></li>
<li class="chapter" data-level="2.1.2" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#content"><i class="fa fa-check"></i><b>2.1.2</b> Scene Content</a></li>
<li class="chapter" data-level="2.1.3" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#hiding-the-hud"><i class="fa fa-check"></i><b>2.1.3</b> Hiding the HUD</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#console"><i class="fa fa-check"></i><b>2.2</b> The Console</a><ul>
<li class="chapter" data-level="2.2.1" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#up-arrow"><i class="fa fa-check"></i><b>2.2.1</b> Up Arrow!</a></li>
<li class="chapter" data-level="2.2.2" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#tangent-on-using-the-command-line-and-the-interactive-menus"><i class="fa fa-check"></i><b>2.2.2</b> Tangent on using the command line and the interactive menus</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#seq"><i class="fa fa-check"></i><b>2.3</b> Sequence Bar</a></li>
<li class="chapter" data-level="2.4" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#Menu"><i class="fa fa-check"></i><b>2.4</b> Menu System</a><ul>
<li class="chapter" data-level="2.4.1" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#top-menu"><i class="fa fa-check"></i><b>2.4.1</b> Top Menu</a></li>
<li class="chapter" data-level="2.4.2" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#toprow"><i class="fa fa-check"></i><b>2.4.2</b> Top row buttons</a></li>
<li class="chapter" data-level="2.4.3" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#pop-up-menu"><i class="fa fa-check"></i><b>2.4.3</b> Pop-up menu</a></li>
<li class="chapter" data-level="2.4.4" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#selmenu"><i class="fa fa-check"></i><b>2.4.4</b> Selection Menus</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#help"><i class="fa fa-check"></i><b>2.5</b> Where to find help</a><ul>
<li class="chapter" data-level="2.5.1" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#searchdoc"><i class="fa fa-check"></i><b>2.5.1</b> SearchDoc</a></li>
<li class="chapter" data-level="2.5.2" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#help-movies"><i class="fa fa-check"></i><b>2.5.2</b> Help Movies</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#files"><i class="fa fa-check"></i><b>2.6</b> File system</a><ul>
<li class="chapter" data-level="2.6.1" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#datam"><i class="fa fa-check"></i><b>2.6.1</b> Naming files and Data Management</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="the-yasara-user-interface.html"><a href="the-yasara-user-interface.html#knowledge-self-checks"><i class="fa fa-check"></i><b>2.7</b> Knowledge Self-Checks</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html"><i class="fa fa-check"></i><b>3</b> File types and how to work with them in YASARA</a><ul>
<li class="chapter" data-level="3.1" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#types"><i class="fa fa-check"></i><b>3.1</b> Common model file types</a><ul>
<li class="chapter" data-level="3.1.1" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#pdb-files"><i class="fa fa-check"></i><b>3.1.1</b> .pdb files</a></li>
<li class="chapter" data-level="3.1.2" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#yob-files-yasara-object"><i class="fa fa-check"></i><b>3.1.2</b> .yob files (YASARA object)</a></li>
<li class="chapter" data-level="3.1.3" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#sce-files-scene-files"><i class="fa fa-check"></i><b>3.1.3</b> .sce files (Scene files)</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#starting-to-work-with-models-in-yasara"><i class="fa fa-check"></i><b>3.2</b> Starting to work with models in YASARA</a></li>
<li class="chapter" data-level="3.3" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#load"><i class="fa fa-check"></i><b>3.3</b> Loading a model</a><ul>
<li class="chapter" data-level="3.3.1" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#pdb-files-1"><i class="fa fa-check"></i><b>3.3.1</b> .pdb files</a></li>
<li class="chapter" data-level="3.3.2" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#getting-a-.pdb-file-from-the-internet"><i class="fa fa-check"></i><b>3.3.2</b> Getting a .pdb file from the internet</a></li>
<li class="chapter" data-level="3.3.3" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#yob-files"><i class="fa fa-check"></i><b>3.3.3</b> .yob files</a></li>
<li class="chapter" data-level="3.3.4" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#sce-files"><i class="fa fa-check"></i><b>3.3.4</b> .sce files</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#what-does-obj-vs.mol-vs.res-vs.atom-mean"><i class="fa fa-check"></i><b>3.4</b> What does Obj vs. Mol vs. Res vs. Atom mean?</a><ul>
<li class="chapter" data-level="3.4.1" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#why-is-the-organization-important"><i class="fa fa-check"></i><b>3.4.1</b> Why is the organization important?</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#build"><i class="fa fa-check"></i><b>3.5</b> Building a molecule</a></li>
<li class="chapter" data-level="3.6" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#save"><i class="fa fa-check"></i><b>3.6</b> Saving Models</a><ul>
<li class="chapter" data-level="3.6.1" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#pdb-files-2"><i class="fa fa-check"></i><b>3.6.1</b> .pdb files</a></li>
<li class="chapter" data-level="3.6.2" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#yob-files-1"><i class="fa fa-check"></i><b>3.6.2</b> .yob files</a></li>
<li class="chapter" data-level="3.6.3" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#sce-files-1"><i class="fa fa-check"></i><b>3.6.3</b> .sce files</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#remove"><i class="fa fa-check"></i><b>3.7</b> Adding and Removing Objects</a></li>
<li class="chapter" data-level="3.8" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#clearing-the-scene"><i class="fa fa-check"></i><b>3.8</b> Clearing the scene</a></li>
<li class="chapter" data-level="3.9" data-path="file-types-and-how-to-work-with-them-in-yasara.html"><a href="file-types-and-how-to-work-with-them-in-yasara.html#knowledge-self-checks-1"><i class="fa fa-check"></i><b>3.9</b> Knowledge Self-Checks</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="manipulate.html"><a href="manipulate.html"><i class="fa fa-check"></i><b>4</b> Manipulating the model</a><ul>
<li class="chapter" data-level="4.1" data-path="manipulate.html"><a href="manipulate.html#orient"><i class="fa fa-check"></i><b>4.1</b> Positioning and orienting the model</a><ul>
<li class="chapter" data-level="4.1.1" data-path="manipulate.html"><a href="manipulate.html#making-reproducible-views"><i class="fa fa-check"></i><b>4.1.1</b> Making Reproducible Views</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="manipulate.html"><a href="manipulate.html#style"><i class="fa fa-check"></i><b>4.2</b> Changing how the model looks</a><ul>
<li class="chapter" data-level="4.2.1" data-path="manipulate.html"><a href="manipulate.html#stylizing-individual-amino-acids"><i class="fa fa-check"></i><b>4.2.1</b> Stylizing individual amino acids</a></li>
<li class="chapter" data-level="4.2.2" data-path="manipulate.html"><a href="manipulate.html#hide"><i class="fa fa-check"></i><b>4.2.2</b> Showing and Hiding Atoms and Amino Acids</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="manipulate.html"><a href="manipulate.html#colors"><i class="fa fa-check"></i><b>4.3</b> Setting the Colors of the Model</a><ul>
<li class="chapter" data-level="4.3.1" data-path="manipulate.html"><a href="manipulate.html#color-wheel"><i class="fa fa-check"></i><b>4.3.1</b> Color Wheel</a></li>
<li class="chapter" data-level="4.3.2" data-path="manipulate.html"><a href="manipulate.html#color-by-element-and-other-features"><i class="fa fa-check"></i><b>4.3.2</b> Color by Element and other features</a></li>
<li class="chapter" data-level="4.3.3" data-path="manipulate.html"><a href="manipulate.html#changing-the-color-scheme"><i class="fa fa-check"></i><b>4.3.3</b> Changing the color scheme</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="manipulate.html"><a href="manipulate.html#surfaces"><i class="fa fa-check"></i><b>4.4</b> Adding a surface</a><ul>
<li class="chapter" data-level="4.4.1" data-path="manipulate.html"><a href="manipulate.html#hiding-a-surface"><i class="fa fa-check"></i><b>4.4.1</b> Hiding a surface</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="manipulate.html"><a href="manipulate.html#knowledge-self-check"><i class="fa fa-check"></i><b>4.5</b> Knowledge Self-Check</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html"><i class="fa fa-check"></i><b>5</b> Adding annotations to your scene</a><ul>
<li class="chapter" data-level="5.1" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#png"><i class="fa fa-check"></i><b>5.1</b> Saving an image</a><ul>
<li class="chapter" data-level="5.1.1" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#background"><i class="fa fa-check"></i><b>5.1.1</b> Change the background color</a></li>
<li class="chapter" data-level="5.1.2" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#hidehud"><i class="fa fa-check"></i><b>5.1.2</b> Hide the HUD!</a></li>
<li class="chapter" data-level="5.1.3" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#rayopt"><i class="fa fa-check"></i><b>5.1.3</b> Ray Tracing options</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#labels"><i class="fa fa-check"></i><b>5.2</b> Labels</a><ul>
<li class="chapter" data-level="5.2.1" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#labpar"><i class="fa fa-check"></i><b>5.2.1</b> Label parameters</a></li>
<li class="chapter" data-level="5.2.2" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#removing-a-label"><i class="fa fa-check"></i><b>5.2.2</b> Removing a label</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#bonds"><i class="fa fa-check"></i><b>5.3</b> Adding Hydrogen Bonds</a><ul>
<li class="chapter" data-level="5.3.1" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#coloring-hydrogen-bonds"><i class="fa fa-check"></i><b>5.3.1</b> Coloring hydrogen bonds</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#figurelessons"><i class="fa fa-check"></i><b>5.4</b> Making good figures</a><ul>
<li class="chapter" data-level="5.4.1" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#considerations-when-making-a-biomolecule-figure"><i class="fa fa-check"></i><b>5.4.1</b> Considerations when making a biomolecule figure</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="adding-annotations-to-your-scene.html"><a href="adding-annotations-to-your-scene.html#knowledge-checks"><i class="fa fa-check"></i><b>5.5</b> Knowledge Checks</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html"><i class="fa fa-check"></i><b>6</b> Making measurements and comparisons</a><ul>
<li class="chapter" data-level="6.1" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#select"><i class="fa fa-check"></i><b>6.1</b> Making Selections</a><ul>
<li class="chapter" data-level="6.1.1" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#making-multiple-selections"><i class="fa fa-check"></i><b>6.1.1</b> Making multiple selections</a></li>
<li class="chapter" data-level="6.1.2" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#how-to-manipulate-selections"><i class="fa fa-check"></i><b>6.1.2</b> How to manipulate selections</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#measure"><i class="fa fa-check"></i><b>6.2</b> Measuring Distance</a><ul>
<li class="chapter" data-level="6.2.1" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#labdist"><i class="fa fa-check"></i><b>6.2.1</b> Labeling distances</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#mangle"><i class="fa fa-check"></i><b>6.3</b> Measuring Angles</a></li>
<li class="chapter" data-level="6.4" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#dihedral"><i class="fa fa-check"></i><b>6.4</b> Measurings Dihedrals</a></li>
<li class="chapter" data-level="6.5" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#count"><i class="fa fa-check"></i><b>6.5</b> Counting Atoms/Residues</a></li>
<li class="chapter" data-level="6.6" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#align"><i class="fa fa-check"></i><b>6.6</b> Alignment</a><ul>
<li class="chapter" data-level="6.6.1" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#why-did-my-objmol-move-or-not"><i class="fa fa-check"></i><b>6.6.1</b> Why did my Obj/Mol move (or not)?</a></li>
<li class="chapter" data-level="6.6.2" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#rmsd"><i class="fa fa-check"></i><b>6.6.2</b> Root Mean Square Deviation</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#swap"><i class="fa fa-check"></i><b>6.7</b> Making amino acid or nucleotide substitutions</a><ul>
<li class="chapter" data-level="6.7.1" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#swapping-bonds"><i class="fa fa-check"></i><b>6.7.1</b> Swapping bonds</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="making-measurements-and-comparisons.html"><a href="making-measurements-and-comparisons.html#knowledge-checks-1"><i class="fa fa-check"></i><b>6.8</b> Knowledge Checks</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="movies.html"><a href="movies.html"><i class="fa fa-check"></i><b>7</b> Macros</a><ul>
<li class="chapter" data-level="7.1" data-path="movies.html"><a href="movies.html#text-editors"><i class="fa fa-check"></i><b>7.1</b> Text editors</a></li>
<li class="chapter" data-level="7.2" data-path="movies.html"><a href="movies.html#macro"><i class="fa fa-check"></i><b>7.2</b> Organization of a macro</a><ul>
<li class="chapter" data-level="7.2.1" data-path="movies.html"><a href="movies.html#the-as-a-way-to-help-your-reader-and-you"><i class="fa fa-check"></i><b>7.2.1</b> The <code>#</code> as a way to help your reader (and you!)</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="movies.html"><a href="movies.html#making-a-macro"><i class="fa fa-check"></i><b>7.3</b> Making a macro</a><ul>
<li class="chapter" data-level="7.3.1" data-path="movies.html"><a href="movies.html#commands-that-can-be-used-in-a-macro"><i class="fa fa-check"></i><b>7.3.1</b> Commands that can be used in a macro</a></li>
<li class="chapter" data-level="7.3.2" data-path="movies.html"><a href="movies.html#macrosave"><i class="fa fa-check"></i><b>7.3.2</b> Saving the macro</a></li>
<li class="chapter" data-level="7.3.3" data-path="movies.html"><a href="movies.html#macroplay"><i class="fa fa-check"></i><b>7.3.3</b> Playing the macro</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="movies.html"><a href="movies.html#macro-examples"><i class="fa fa-check"></i><b>7.4</b> Macro examples</a></li>
<li class="chapter" data-level="7.5" data-path="movies.html"><a href="movies.html#knowledge-checks-2"><i class="fa fa-check"></i><b>7.5</b> Knowledge Checks</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="exercise-answers.html"><a href="exercise-answers.html"><i class="fa fa-check"></i><b>8</b> Exercise Answers</a><ul>
<li class="chapter" data-level="8.1" data-path="exercise-answers.html"><a href="exercise-answers.html#threekey"><i class="fa fa-check"></i><b>8.1</b> Chapter 3</a></li>
<li class="chapter" data-level="8.2" data-path="exercise-answers.html"><a href="exercise-answers.html#fourkey"><i class="fa fa-check"></i><b>8.2</b> Chapter 4</a></li>
<li class="chapter" data-level="8.3" data-path="exercise-answers.html"><a href="exercise-answers.html#fivekey"><i class="fa fa-check"></i><b>8.3</b> Chapter 5</a></li>
<li class="chapter" data-level="8.4" data-path="exercise-answers.html"><a href="exercise-answers.html#sixkey"><i class="fa fa-check"></i><b>8.4</b> Chapter 6</a></li>
<li class="chapter" data-level="8.5" data-path="exercise-answers.html"><a href="exercise-answers.html#sevenkey"><i class="fa fa-check"></i><b>8.5</b> Chapter 6</a></li>
<li class="chapter" data-level="8.6" data-path="exercise-answers.html"><a href="exercise-answers.html#eightkey"><i class="fa fa-check"></i><b>8.6</b> Chapter 7</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="manage.html"><a href="manage.html"><i class="fa fa-check"></i><b>9</b> Data Management</a><ul>
<li class="chapter" data-level="9.1" data-path="manage.html"><a href="manage.html#introduction-to-the-problem"><i class="fa fa-check"></i><b>9.1</b> Introduction to the problem</a></li>
<li class="chapter" data-level="9.2" data-path="manage.html"><a href="manage.html#what-are-data"><i class="fa fa-check"></i><b>9.2</b> What are Data?</a></li>
<li class="chapter" data-level="9.3" data-path="manage.html"><a href="manage.html#effective-file-naming"><i class="fa fa-check"></i><b>9.3</b> Effective File naming</a><ul>
<li class="chapter" data-level="9.3.1" data-path="manage.html"><a href="manage.html#information-to-consider-including-in-file-names"><i class="fa fa-check"></i><b>9.3.1</b> Information to consider including in file names</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="manage.html"><a href="manage.html#data-and-file-naming-dictionary"><i class="fa fa-check"></i><b>9.4</b> Data and File Naming Dictionary</a></li>
<li class="chapter" data-level="9.5" data-path="manage.html"><a href="manage.html#data-management-plan"><i class="fa fa-check"></i><b>9.5</b> Data Management Plan</a><ul>
<li class="chapter" data-level="9.5.1" data-path="manage.html"><a href="manage.html#data-management-plan-questions-for-students-in-biochemistry-class"><i class="fa fa-check"></i><b>9.5.1</b> Data Management Plan questions for students in Biochemistry class:</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html"><i class="fa fa-check"></i><b>10</b> Errors and quirks</a><ul>
<li class="chapter" data-level="10.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#macrotext"><i class="fa fa-check"></i><b>10.1</b> .mcr –&gt; .txt and back again</a><ul>
<li class="chapter" data-level="10.1.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#fix"><i class="fa fa-check"></i><b>10.1.1</b> Fix</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#running-in-the-downloads-folder"><i class="fa fa-check"></i><b>10.2</b> Running in the downloads folder</a><ul>
<li class="chapter" data-level="10.2.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#fix-1"><i class="fa fa-check"></i><b>10.2.1</b> Fix</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#admin-permission-needed"><i class="fa fa-check"></i><b>10.3</b> Admin permission needed</a><ul>
<li class="chapter" data-level="10.3.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#fix-2"><i class="fa fa-check"></i><b>10.3.1</b> Fix</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#macos-wont-allow-yasara-to-run-because-not-from-app-store"><i class="fa fa-check"></i><b>10.4</b> MacOS won’t allow YASARA to run because not from App Store</a><ul>
<li class="chapter" data-level="10.4.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#fix-3"><i class="fa fa-check"></i><b>10.4.1</b> Fix</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#menu-bar-disappears"><i class="fa fa-check"></i><b>10.5</b> Menu bar disappears</a><ul>
<li class="chapter" data-level="10.5.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#fix-4"><i class="fa fa-check"></i><b>10.5.1</b> Fix</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#menu-bar-or-continue-button-does-not-respond-to-clicks"><i class="fa fa-check"></i><b>10.6</b> Menu bar or Continue Button does not respond to clicks</a><ul>
<li class="chapter" data-level="10.6.1" data-path="errors-and-quirks.html"><a href="errors-and-quirks.html#fix-5"><i class="fa fa-check"></i><b>10.6.1</b> Fix</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">
Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">A Guide to YASARA</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="making-measurements-and-comparisons" class="section level1">
<h1><span class="header-section-number">Chapter 6</span> Making measurements and comparisons</h1>
<p>In addition to being a good program for visualizing biomolecules, YASARA can also be a tool to analyze biomolecule structure. In the free version the analysis capabilities are limited, however for most biomolecules, these simple tools are sufficient.</p>
<div id="select" class="section level2">
<h2><span class="header-section-number">6.1</span> Making Selections</h2>
<p>All analytical methods in YASARA require a definition of the Obj/Mol/Res/Atoms involved. Selections can be made using the mouse as described in Table <a href="manipulate.html#tab:mousetable">4.1</a> or using the <code>Select</code> command in the menu or command line. Once a selection has been made with the mouse you will see “fireflies” (<a href="making-measurements-and-comparisons.html#fig:fireflies">6.1</a>) around the atom selected.</p>
<div class="figure"><span id="fig:fireflies"></span>
<img src="images/fireflies.PNG" alt="A selected atom surrounded by fireflies shown as white streaks" width="100%" />
<p class="caption">
Figure 6.1: A selected atom surrounded by fireflies shown as white streaks
</p>
</div>
<p>In contrast, selections made using the <code>Select</code> command will become highlighted with in a universe style (<a href="making-measurements-and-comparisons.html#fig:uni">6.2</a>).</p>
<div class="figure"><span id="fig:uni"></span>
<img src="images/seluniverse.PNG" alt="A selected atom with the universe style. It's Full of Stars" width="100%" />
<p class="caption">
Figure 6.2: A selected atom with the universe style. It’s Full of Stars
</p>
</div>
<p>By far, the easier way to make single selections is using the mouse. However, when making selections of groups such as all the alanine residues or multiple selections, then you will need the menu (<a href="the-yasara-user-interface.html#selmenu">2.4.4</a>) or command line (<a href="making-measurements-and-comparisons.html#selcmd">6.1.1.4</a>).</p>
<div id="making-multiple-selections" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Making multiple selections</h3>
<div id="using-the-mouse" class="section level4">
<h4><span class="header-section-number">6.1.1.1</span> Using the Mouse</h4>
<p>To select multiple atoms using the mouse, hold down the CTRL key while left clicking the target atoms. As each atom is selected a new set of fireflies will appear around the atom. The first atom will always have white fireflies while the subsequent selections will have red, green, or blue fireflies. Red fireflies always designate the last atom selected. This method of selecting atoms is how we will make measurements of distance (<a href="making-measurements-and-comparisons.html#measure">6.2</a>) and angles (@ref(#mangle)).</p>
</div>
<div id="selmenu" class="section level4">
<h4><span class="header-section-number">6.1.1.2</span> Using Select menu</h4>
<p>The select menu offers several methods to make selections which each have advantages and quirks.</p>
<ul>
<li><p><strong>New</strong> – This method is similar to the ways we have worked through a menu tree previously. When you do Select&gt;New&gt;<em>Make your selection</em>, you enter into the standard menu system of YASARA. If you want to make multiple selections, you can left click and hold down the CTRL key. Alternatively, left click and hold down the SHIFT key to select consecutive lines in the menu. Once the selection is made, they will be come highlighted in the universe style (<a href="making-measurements-and-comparisons.html#fig:uni">6.2</a>).</p></li>
<li><p><strong>Additional</strong> – This method <em>adds</em> to an <em>existing</em> selection.</p></li>
<li><p><strong>In box</strong> – This method changes the cursor and upon left-click draws a red and yellow box which you can drag to expand. Any Obj/Mol/Res/Atoms within the box that you drew become selected. One caution is that you cannot change the orientation of the scene once you are in this mode, so have the molecule set and then do this method. An additional caution is that everything that can be seen within the box is selected, so you may inadvertantly select Obj/Mol/Res/Atoms you did not intend. After drawing the box, selections are highlighted in the universe style (<a href="making-measurements-and-comparisons.html#fig:uni">6.2</a>). This method also can be accessed through the top row buttons (<a href="the-yasara-user-interface.html#toprow">2.4.2</a>).</p></li>
<li><p><strong>In arbitrary area</strong> – This is similar to the lasso tool used in other programs. As with <em>In box</em>, this method changes the cursor so that you can draw a roughly circular shape around the amino acids to be selected. All the cautions of <em>In box</em> also exist for this method. After drawing the box, selections are highlighted in the universe style (<a href="making-measurements-and-comparisons.html#fig:uni">6.2</a>). This method also can be accessed through the top row buttons (<a href="the-yasara-user-interface.html#toprow">2.4.2</a>).</p></li>
<li><p><strong>In sphere around</strong> – This method is one of the more interesting and useful ways to make a selection, especially if you want to analyze or style the environment around a specific amino acid. Upon choosing this option, you will be asked to choose the Obj/Mol/Res/Atom that you want to draw the sphere around, and then you can interactively size the sphere. On screen is also a guide showing the radius of the sphere in angstroms. After drawing the box, selections are highlighted in the universe style (<a href="making-measurements-and-comparisons.html#fig:uni">6.2</a>).</p></li>
<li><p><strong>Invert</strong> – This method changes an existing selection to select everything that was not previously selected. This can be useful for styling or analyzing two regions of the molecule separately.</p></li>
</ul>
</div>
<div id="unselect" class="section level4">
<h4><span class="header-section-number">6.1.1.3</span> Unselect</h4>
<p>In addition to the <code>Select</code> command, there is an <code>UnSelect</code> command which is good for clearing selections or removing certain elements from your selection. Typically I only use <code>UnSelectAll</code>. However, <code>UnSelect</code> combined with Obj/Mol/Res/Atom option can be useful in macros (<a href="#macros"><strong>??</strong></a>) and movies (<a href="movies.html#movies">7</a>) for creating interesting styles and views of the molecule.</p>
</div>
<div id="selcmd" class="section level4">
<h4><span class="header-section-number">6.1.1.4</span> Using <code>Select</code> from the command line</h4>
<p>Select from the command line involves using <code>Select</code> in combination with a level of organization followed by a selection. This method is fast and precise in addition to being useful for manipulation and styling scenes quickly.</p>
<p>You can also add on qualifiers to select such as <code>with distance &lt;5 from Res 10</code> which would select all the amino acids within 5 <span class="math inline">\(\unicode{x212B}\)</span> of amino acid 10 in the protein. Also you can use <code>!</code> to mean not. The command <code>!lys</code> would refer to everything except the lysine amino acids in the scene.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Select all the alanine residues in the protein</span>
<span class="op">&gt;</span>SelectRes ala

<span class="co"># Select amino acids 1-10 inclusive</span>
<span class="op">&gt;</span>SelectRes <span class="dv">1</span><span class="op">-</span><span class="dv">10</span>

<span class="co"># Select molecule A</span>
<span class="op">&gt;</span>SelectMol A

<span class="co"># Select amino acids within 4 angstroms of amino acid 10</span>
<span class="op">&gt;</span>SelectRes all with distance <span class="op">&lt;</span><span class="dv">4</span>

<span class="co"># Select all amino acids except alanine</span>
<span class="op">&gt;</span>SelectRes <span class="op">!</span>ala</code></pre></div>
</div>
</div>
<div id="how-to-manipulate-selections" class="section level3">
<h3><span class="header-section-number">6.1.2</span> How to manipulate selections</h3>
<p>Once a selection has been made using the various methods above, what can you do with that? Selections like Obj or Res can be referred to in commands to produce analysis or changes to the scene. Previously if we wanted to change the style of an amino acid to ball and stick we typed <code>BallStickRes</code> followed by the amino acid number or name. Now we can insert <code>selected</code> in for the amino acid name or number as shown in the code chunk below. While this seems redundant, if you have a complex set of amino acids that you are interested in. Typing that list for every command would be tedious. Instead, <code>Select</code> those amino acids and then refer that selection. This trick can save you a lot of time.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Select all the alanine residues in the protein</span>
<span class="op">&gt;</span>SelectRes ala

<span class="co"># Change the style of all the alanines to ball and stick</span>
<span class="op">&gt;</span>BallStickRes selected

<span class="co"># Color the alanines red</span>
<span class="op">&gt;</span>ColorRes selected, red

<span class="co"># Show an individual surface around each of the alanines</span>
<span class="op">&gt;</span>ShowSurfRes Selected,Type=Molecular,OutCol=Blue,OutAlpha=<span class="dv">50</span>,Specular=Yes</code></pre></div>
</div>
</div>
<div id="measure" class="section level2">
<h2><span class="header-section-number">6.2</span> Measuring Distance</h2>
<p>The interval between atoms is crucial for understanding strong and weak interactions in chemistry and biology<a href="#fn7" class="footnoteRef" id="fnref7"><sup>7</sup></a>. In YASARA the command to measure this interval is <code>Distance</code> but you can also use the menu system or the mouse. This latter method is by far the easiest.</p>
<p>Using the mouse:</p>
<p>Left-click on the first atom and then hold down the CTRL key and left-click on the second atom. Fireflies will appear around both selections and the distance will appear under Marked Distance in the HUD (<a href="the-yasara-user-interface.html#HUD">2.1</a>). The units of distance are <span class="math inline">\(\unicode{x212B}\)</span> or Angstrom, which is equal to 10<sup>-10</sup> meters.</p>
<p>Using the menu:</p>
<p>Analyze&gt;Geometry&gt;Distance, then select the two atoms you are want to measure the interval between. There is an option to include only the distance between atoms that are directly connected, however this is not useful in most cases.</p>
<p>Using the command line:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Measure the distance between the Oxygen in the side chain of Threonine 2</span>
<span class="co"># and the nitrogen in the side chain of Arginine 10</span>
<span class="op">&gt;</span>Distance <span class="dv">13</span>,<span class="dv">69</span>,bound=No
<span class="co"># Output from YASARA in the console</span>
Distance  <span class="dv">13</span>  OG1 THR    <span class="dv">2</span>    <span class="op">-</span><span class="st">  </span><span class="dv">69</span>  NH1 ARG   <span class="dv">10</span>   <span class="op">:</span><span class="st">     </span><span class="fl">2.897</span> A</code></pre></div>
<p>In most cases, using the mouse is more than sufficient unless you want to use a Macro to analyse the distances of a large number of atoms.</p>
<div id="labdist" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Labeling distances</h3>
<p>For annotating distances for a figure, you would use the command <code>LabelDis</code> or choose to label a distance in the label menu (see section <a href="adding-annotations-to-your-scene.html#labels">5.2</a> for more details on annotating scenes with labels). I will only desribe the command line method here with the results of the command shown in Figure <a href="making-measurements-and-comparisons.html#fig:labeldisfig">6.3</a>. As with text labels, when labeling the distance you need to specify the text to show (format), the size (height), color (color), and the coordinates of the label relative to the <em>center</em> of the gap between the atoms. As with the <code>Distance</code>, you need to specify the atoms to measure between and bound.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Measure the distance between the Oxygen in the side chain of Threonine 2</span>
<span class="co"># and the nitrogen in the side chain of Arginine 10 and add a label</span>
<span class="op">&gt;</span>LabelDis <span class="dv">13</span>,<span class="dv">69</span>,Format=DIS,Height=<span class="fl">0.7</span>,Color=Black,X=<span class="fl">0.0</span>,Y=<span class="fl">0.0</span>,Z=<span class="fl">0.0</span>,bound=No

<span class="co">#Orient the scene to show the label</span>
<span class="op">&gt;</span>PosOriObj <span class="dv">01</span>, X=<span class="fl">0.000</span>, Y=<span class="fl">0.000</span>, Z=<span class="fl">35.192</span>, Alpha=<span class="fl">31.934</span>, Beta=<span class="fl">292.437</span>, Gamma=<span class="op">-</span><span class="fl">9.577</span></code></pre></div>
<div class="figure"><span id="fig:labeldisfig"></span>
<img src="images/labeldis.png" alt="Annotated distance between Threonine 2 and Arginine 10" width="100%" />
<p class="caption">
Figure 6.3: Annotated distance between Threonine 2 and Arginine 10
</p>
</div>
</div>
</div>
<div id="mangle" class="section level2">
<h2><span class="header-section-number">6.3</span> Measuring Angles</h2>
<p>Angles refers to the distance between two intersecting bonds or imaginary lines between three Obj/Mol/Res/Atom. Like <code>Distance</code>, there are three methods to calculate this value, however three Obj/Mol/Res/Atoms are needed rather than two. I do not use the <code>Angle</code> command frequently except to check that <a href="https://en.wikipedia.org/wiki/Molecular_geometry">bond angles</a> are accurate when building models.</p>
<p>Using the mouse:</p>
<p>Left-click on the first atom and then hold down the CTRL key and left-click on the second, third atoms. Fireflies will appear around all selections and the angle will appear under Marked Angle along with the distance between the first two selections under Marked Distance in the HUD (<a href="the-yasara-user-interface.html#HUD">2.1</a>). The units of angle are <sup><span class="math inline">\(\circ\)</span></sup> or degrees.</p>
<p>Using the menu:</p>
<p>Analyze&gt;Geometry&gt;Angle, then select the three atoms you are want to measure the angle between There is an option to include only the distance between atoms that are directly connected, however this is not useful in most cases.</p>
<p>Using the command line:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Measure the angle between the Oxygen in the side chain of Threonine 2</span>
<span class="co"># and the nitrogen in the side chain of Arginine 10 and the carbon in the sidechain</span>
<span class="op">&gt;</span>Angle <span class="dv">68</span>,<span class="dv">69</span>,<span class="dv">13</span>,bound=No
<span class="co"># Output from YASARA in the console</span>
Angle  <span class="dv">68</span>  CZ  ARG   <span class="dv">10</span>    <span class="op">-</span><span class="st">  </span><span class="dv">69</span>  NH1 ARG   <span class="dv">10</span>    <span class="op">-</span><span class="st">  </span><span class="dv">13</span>  OG1 THR    <span class="dv">2</span>   <span class="op">:</span><span class="st">  </span><span class="fl">124.870</span></code></pre></div>
<p>There is no way to label an angle in YASARA at this point in time, however it would be possible to show using a regular label as described in the section on labeling (<a href="adding-annotations-to-your-scene.html#labels">5.2</a>).</p>
</div>
<div id="dihedral" class="section level2">
<h2><span class="header-section-number">6.4</span> Measurings Dihedrals</h2>
<p>The Dihedral angle is the angle between two planes created by two attached atoms and therefore involves 4 atoms. The most common dihedral angles to measure in proteins are the <a href="http://proteopedia.org/wiki/index.php/Dihedral/Index">Ramachandran Angles</a>. There are a limited number of stable dihedral angles in biomolecules due to steric clashes, therefore an easy way to check a protein structure is to measure the dihedral angles. To measure dihedral angle requires 4 atom selections.</p>
<p>Using the mouse:</p>
<p>Left-click on the first atom and then hold down the CTRL key and left-click on the second, third, and fourth atoms. Fireflies will appear around all selections and the dihedral angle will appear under Marked Dihedral and the angle of the first, third, and fourth atoms under Marked Angle along with the distance between the firsth and fourth selections under Marked Distance in the HUD (<a href="the-yasara-user-interface.html#HUD">2.1</a>). The units of the dihedral angle are <sup><span class="math inline">\(\circ\)</span></sup> or degrees.</p>
<p>Using the menu:</p>
<p>Analyze&gt;Geometry&gt;Dihedral, then select the four atoms you are want to measure the angle between. There is an option to include only the distance between atoms that are directly connected, however this is not useful in most cases.</p>
<p>Using the command line:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Measure the dihedral angle beta carbon, alpha carbon, carbonyl carbon, and carbonyl</span>
<span class="co"># oxygen of Arginine 10. This is also the Phi angle for this amino acid.</span>
<span class="op">&gt;</span>Dihedral <span class="dv">64</span>,<span class="dv">61</span>,<span class="dv">62</span>,<span class="dv">63</span>,bound=No
<span class="co"># Output from YASARA in the console</span>
Dihedral  <span class="dv">64</span>  CB  ARG   <span class="dv">10</span>    <span class="op">-</span><span class="st">  </span><span class="dv">61</span>  CA  ARG   <span class="dv">10</span>    <span class="op">-</span><span class="st">  </span><span class="dv">62</span>  C   ARG   <span class="dv">10</span>    <span class="op">-</span><span class="st">  </span><span class="dv">63</span>  O   ARG   <span class="dv">10</span>   <span class="op">:</span><span class="st">  </span><span class="op">-</span><span class="fl">99.067</span></code></pre></div>
<p>There is no way to label a dihedral angle in YASARA at this point in time, however it would be possible to show using a regular label as described in the section on labeling (<a href="adding-annotations-to-your-scene.html#labels">5.2</a>).</p>
</div>
<div id="count" class="section level2">
<h2><span class="header-section-number">6.5</span> Counting Atoms/Residues</h2>
<p>The <code>Count</code> command is for quickly counting the number of Obj/Mol/Res/Atoms or bonds in a scene. This command can also be mixed to determine the number of Res/atoms in all of certain Obj/Mol/Res or within a selection.</p>
<p>Using the menu:</p>
<p>Analyze&gt;Count, then select Obj/Mol/Res/Atom or bond. In the menus, then make your selection. I recommend using the Name or Belonds to or has boxes in most cases (see Section <a href="the-yasara-user-interface.html#selmenu">2.4.4</a> for more details).</p>
<p>Using the command line:</p>
<p>The primary command is <code>Count</code> followed by Obl/Mol/Res/Atom/Bond and a selection. The number of things that match that selection will be shown in the console.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Count the cysteines in the protein</span>
<span class="op">&gt;</span>CountRes cys

<span class="co"># Count the atoms in the scene</span>
<span class="op">&gt;</span>CountAtom all
<span class="co"># or</span>
<span class="op">&gt;</span>CountAtom Obj <span class="dv">1</span>

<span class="co"># Count the atoms in the cysteines </span>
<span class="op">&gt;</span>CountAtom Res Cys

<span class="co"># Count the atoms in the backbone of the cysteine</span>
<span class="op">&gt;</span>CountAtom Res Cys backbone</code></pre></div>
</div>
<div id="align" class="section level2">
<h2><span class="header-section-number">6.6</span> Alignment</h2>
<p>When trying to understand the function of biomolecules it is often useful to compare the structure of the two molecules after alignment. There are two functions to align Obj/Mols in YASARA, Superpose (<code>Sup</code>) and <code>Align</code>. Each has its own unique style. <code>Sup</code> aligns molecules using sequence while <code>Align</code> uses structure. The end result of either action should be two or more Obj/Mols shown on top of each other along with a measure of the similarity called the Root Mean Square Deviation (RMSD) which is described more in the next section (<a href="making-measurements-and-comparisons.html#rmsd">6.6.2</a>). In general, <code>Align</code> is the better option to use, but <code>Sup</code> can often work when <code>Align</code> fails.</p>
<p>To superpose two or more objects/molecules from the menu:</p>
<p>Analyze&gt;Superpose&gt;then pick whether you want to superpose on a selection (most common) or on the ordered regions. I recommend using <code>SearchDoc</code> to get more details on <code>Sup</code> if you need them. If you choose Superpose&gt;individual objects on selected, you will then be asked to select the Obj/Mol as we have seen before in selection menus (<a href="the-yasara-user-interface.html#selmenu">2.4.4</a>) followed by a set of parameters. Here you will choose what properties must match for the superposition to occur. If you have distinct sequences, then atom name may be the only match, more similar sequences you can probably include residue name and number. Because some bonds can rotate, an additional option is to flip chemically equivalent groups. This parameter is especially useful when you are comparing distinct structures of the same protein. Finally, you can have YASARA return RMSD value per Obj, Mol, Residue or Atom. For distinct proteins, choose Obj or Mol, for similar proteins choose Residue or Atom.</p>
<p>To align two or more objects/molecules from the menu:</p>
<p>Analyze&gt;Align&gt;then pick your method pairwise based on sequence or structure, or multiple based on structure. Once, you pick your method you will be asked to select the Obj/Mol to align using the selection menus (<a href="the-yasara-user-interface.html#selmenu">2.4.4</a>). If doing pairwise, only pick one Obj/Mol per menu, if doing multiple, then you can pick many structures in the first menu followed by the target of the alignment in the second menu.</p>
<p>To align or superpose using the command line:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load two PDB files</span>
<span class="op">&gt;</span>LoadPDB 1ubq, download=yes
<span class="op">&gt;</span>LoaDPDB 5ia7, download=yes

<span class="co"># Superpose 5ia7 (Obj 2) on 1ubq (Obj 1) and return RMSD values of the Obj</span>
<span class="op">&gt;</span>SupObj <span class="dv">2</span>,<span class="dv">1</span>,Match=AtomName,Flip=No,Unit=Obj
<span class="co"># Console output</span>
Object  <span class="dv">2</span> (5ia7) and Object  <span class="dv">1</span> (1ubq) have <span class="fl">13.5494</span> A RMSD over <span class="dv">163</span> matched atoms

<span class="co"># Align 5ia7 (Obj 2) on 1ubq (Obj 1) and return RMSD values</span>
<span class="op">&gt;</span>AlignObj <span class="dv">2</span>,<span class="dv">1</span>,Method=MUSTANGPP
<span class="co"># Console output</span>
Single line alignment<span class="op">:</span>
<span class="er">&gt;</span>Object <span class="dv">1</span> (1ubq)<span class="op">:</span>
MQIFVKTL<span class="op">-----</span>TGKTITLEVEPSDTIENVKAKIQDKEGIPPDQQRLI<span class="op">-----</span>FAGKQLEDGRTLSDY<span class="op">------</span>NIQKESTLHLV<span class="op">--------------------------------------------------------------------------------</span>LRLRGG
<span class="op">&gt;</span>Object <span class="dv">2</span> (5ia7)<span class="op">:</span>
VSFKITLTSDPRL<span class="op">--</span>PYKVLSVPESTPFTAVLKFAAEEFKVPAATSAIITNDGI<span class="op">----</span>GINPAQTAGNVFLKHGS<span class="op">------</span>ELRII<span class="op">|</span>VSFKITLTSDPRLPYKVLSVPESTPFTAVLKFAAEEFKVPAATSAIITNDGIGINPAQTAGNVFLKHGSELRIIPRDRV<span class="op">------</span>

Multi line alignment of object <span class="dv">1</span> (1ubq,Query) and object <span class="dv">2</span> (5ia7,Hit)<span class="op">:</span>

Query<span class="op">:</span><span class="st">    </span><span class="dv">1</span> MQIFVKTLTG<span class="op">-----</span>KTITLEVEPSDTIENVKAKIQDKEGIPPDQQRLIFAGK<span class="op">-----</span>QLEDGRTLSDYNIQKES<span class="op">------</span>TLHLVLRLRGG<span class="op">--------------------------------------------------------------------------------</span><span class="st"> </span><span class="dv">77</span>          
            <span class="op">:::</span><span class="st"> </span><span class="er">|</span><span class="st">             </span><span class="er">:</span>L<span class="op">:</span>V  S  <span class="op">:</span><span class="st">  </span>V     <span class="op">|</span><span class="er">:</span><span class="st">  </span><span class="er">|</span>P   <span class="op">:</span><span class="st"> </span><span class="er">|</span>I          <span class="op">|</span><span class="er">:</span><span class="st"> </span><span class="er">::</span>T <span class="op">::</span><span class="st">              </span>L<span class="op">:</span><span class="er">||</span><span class="st">                                                                                                   </span>
Hit  <span class="op">:</span><span class="st">    </span><span class="dv">1</span> VSFKITLT<span class="op">--</span>SDPRLPYKVLSVPESTPFTAVLKFAAEEFKVPAATSAII<span class="op">----</span>TNDGIGINPAQTAGNV<span class="op">------</span>FLKHGSELRII<span class="op">------</span><span class="er">|</span>VSFKITLTSDPRLPYKVLSVPESTPFTAVLKFAAEEFKVPAATSAIITNDGIGINPAQTAGNVFLKHGSELRIIPRDRV <span class="dv">154</span>         
                                                                                                                                                                                                     


The above structural alignment between objects  <span class="dv">2</span> (5ia7) and  <span class="dv">1</span> (1ubq) has an RMSD of <span class="fl">1.629</span> A over <span class="dv">58</span> aligned residues with <span class="fl">13.79</span>% sequence identity.</code></pre></div>
<p><code>Align</code> provides a sequence alignment of the protein in the console in addition to the RMSD value. The query is the second molecule in the command, the Hit is the first molecule. Between the two lines are either a gap, an amino acid in one letter code, a <strong>:</strong> or a <strong>|</strong>. The : indicates similar amino acid based side/charge, | means near identity, and if a letter is shown it means that the amino acid is conserved or the same in that position in both molecules.</p>
<p>As noted above, <code>Align</code> is a more robust method, provides more information, and is simpler to do, however it does not always work, especially for structurally divergent molecules or for very large complex biomolecules.</p>
<div id="why-did-my-objmol-move-or-not" class="section level3">
<h3><span class="header-section-number">6.6.1</span> Why did my Obj/Mol move (or not)?</h3>
<p>When aligning or superposing Obj/Mol, one or more of the Obj/Mols will move on top of another Obj/Mol. At first, this change in the scene can appear confusing and/or frustrating. When setting up an alignment, whatever Obj/Mol(s) is/are listed <em>first</em> will move, while the “target” molecule will remain fixed on the screen. In practice, always have the biomolecule which is better established or which has the longer sequence be the target of the alignment. If molecules disappear you can use the center button in the top row menu (<a href="the-yasara-user-interface.html#toprow">2.4.2</a>) to bring all the molecules back on the screen.</p>
</div>
<div id="rmsd" class="section level3">
<h3><span class="header-section-number">6.6.2</span> Root Mean Square Deviation</h3>
<p>The RMSD provides a measure of the average difference in distance between atoms within two or more molecules. The more closely the structures match, the lower the RMSD value<a href="#fn8" class="footnoteRef" id="fnref8"><sup>8</sup></a>. If you align one or more molecules using <code>Align</code> or <code>Sup</code> this value is provided in the console. You can also calculate RMSD without aligning using the <code>RMSD</code> command followed by Obj/Mol/Res/Atom. This command is useful for calculating the RMSD of pieces of aligned molecules. The <code>RMSD</code> command is found in the Analyze menu and uses a selection menu system as described before (<a href="the-yasara-user-interface.html#selmenu">2.4.4</a>).</p>
</div>
</div>
<div id="swap" class="section level2">
<h2><span class="header-section-number">6.7</span> Making amino acid or nucleotide substitutions</h2>
<p>A frequent goal of biochemistry is to understand biomolecule structure and function. Along the path to this goal, biochemists often like to make substitutions within the sequence of DNA/RNA or proteins to determine how this change alters the structure and function of the biomolecule. YASARA can show simple changes to biomolecule structure using the <code>Swap</code> command which, when combined with energy minimization, molecular dynamics, or experimental data, can explain the structural basis for defects in protein function.</p>
<p><strong>One caution</strong> when using YASARA_view_, <code>Swap</code> simply inserts the amino acid/nucleotide and does not optimize the interactions or any parameters of the changed molecule. Thus, any data associated with ‘Swap’ should be considered with skepticism in the absence of confirmatory evidence.</p>
<p>Using the menu:</p>
<p>Edit&gt;Swap&gt;then select Obj/Mol/Res/Atom or bond. In the next menu, pick your amino acid or nucleotide to change and click OK. In the next menu, choose what to change your selection to. In humans and other living, use L-amino acids. You can initially select multiple amino acids to change, but can only choose one type of amino to change those amino acids/nucleotides to.</p>
<p>Using the pop-up menu:</p>
<p>Select an atom by left-clicking the mouse, then right-click that selected atom. In the pop-up menu, there is the <code>Swap</code> command, which can then be followed as above to change that particular amino acid/nucleotide.</p>
<p>Using the command line:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Load a PDB file</span>
<span class="op">&gt;</span>LoadPDB 1crn, download=yes

<span class="co"># Change Cys 16 in the molecule to Arginine</span>
<span class="op">&gt;</span>SwapRes Cys <span class="dv">16</span>, Arg

<span class="co"># Change all the Asn to Asp</span>
<span class="op">&gt;</span>SwapRes Asn, Asp</code></pre></div>
<div id="swapping-bonds" class="section level3">
<h3><span class="header-section-number">6.7.1</span> Swapping bonds</h3>
<p>During model building of enzymes, it is often useful to build intermediates or transition states of the reactants in order to predict the enzyme active site might interact with these structures. Like for amino acids and nucleotides, <code>Swap</code> can be used to change a double bond into a single bond or other combinations to build these intermediates. Like residue substitution, these changes to bonding are not optimized and should be interpreted cautiously.</p>
</div>
</div>
<div id="knowledge-checks-1" class="section level2">
<h2><span class="header-section-number">6.8</span> Knowledge Checks<a href="#fn9" class="footnoteRef" id="fnref9"><sup>9</sup></a></h2>
<ol style="list-style-type: decimal">
<li><p>Load the PDB file <a href="https://www.rcsb.org/structure/148l">148L</a> and count the number of water molecules around the protein. The residue code for water is hoh.</p></li>
<li><p>Select and count the amino acids within the active site of the protein that are within 3 <span class="math inline">\(\unicode{x212B}\)</span> of Mol S. You will likely need to exclude the waters for this, either by deleting the waters or by being very precise in your selection commands.</p></li>
<li></li>
</ol>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="7">
<li id="fn7"><p>See the section on <a href="https://www.ncbi.nlm.nih.gov/books/NBK22567/">Chemical Bonds in Biochemistry</a> described within 5<sup>th</sup> edition of Biochemistry by JM Berg, JL Tymoczko, and L Stryer.<a href="making-measurements-and-comparisons.html#fnref7">↩</a></p></li>
<li id="fn8"><p>more information on <a href="https://en.wikipedia.org/wiki/Root-mean-square_deviation_of_atomic_positions">RMSD</a> and on <a href="http://proteopedia.org/wiki/index.php/Structural_alignment_tools">programs to calculate RMSD with biomolecules</a><a href="making-measurements-and-comparisons.html#fnref8">↩</a></p></li>
<li id="fn9"><p>Answers can be found <a href="exercise-answers.html#sevenkey">8.5</a><a href="making-measurements-and-comparisons.html#fnref9">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="adding-annotations-to-your-scene.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="movies.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"],
"instapper": false
},
"fontsettings": {
"theme": "white",
"family": "serif",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
},
"search": true,
"toolbar": {
"position": "fixed"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
